### PR TITLE
Implement: -XX:+UseZGC is currently unsupported for -XX:CRaCCheckpointTo

### PR DESCRIPTION
**Pull request by jankratochvil**
---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Radim Vansa](https://openjdk.org/census#rvansa) (@rvansa - Committer)
 * [Anton Kozlov](https://openjdk.org/census#akozlov) (@AntonKozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/crac.git pull/90/head:pull/90` \
`$ git checkout pull/90`

Update a local copy of the PR: \
`$ git checkout pull/90` \
`$ git pull https://git.openjdk.org/crac.git pull/90/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 90`

View PR using the GUI difftool: \
`$ git pr show -t 90`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/crac/pull/90.diff">https://git.openjdk.org/crac/pull/90.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/crac/pull/90#issuecomment-1636735539)

### CODE REVIEW

**Comment by jankratochvil** [comment]
According to @rvansa I should not refer to Azul internal bugs in CRaC branch upstream so I will have to move the internal bug to Github first. I will reopen it afterwards.

File: src/hotspot/share/runtime/arguments.cpp [hunk 1]
```diff
@@ -3128,6 +3128,13 @@ jint Arguments::finalize_vm_init_args(bool patch_mod_javabase) {
     return JNI_ERR;
   }
 
+  if (CRaCCheckpointTo && UseZGC) {
+    # jdk.crac.impl.CheckpointOpenFileException: FD fd=3 type=regular path=/memfd:java_heap (deleted)
```
Comment (rvansa) [inline src/hotspot/share/runtime/arguments.cpp:3132]: Is this meant to be a comment or preprocessor directive?

**Comment by jankratochvil** [comment]
It does not build due to assertion, I have to fix that.

**Comment by jankratochvil** [comment]
That was only a local testing error.

**Review by rvansa — APPROVED** [review]

**Review by AntonKozlov — APPROVED** [review]
LGTM, thank you!

**Pull request closed without merging**